### PR TITLE
Support number as field name

### PR DESCRIPTION
**Pull request by XadillaX**
E.g.

```sql
CREATE TABLE `test` (
  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
  `1` int(11) NOT NULL,
  `2` int(11) NOT NULL,
  `3` int(11) NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
```

And my code would be like:

```javascript
let id = 5;
let idx = 1;
let sql = "UPDATE `test` SET ?? = ?? + (?) WHERE `id` = ?";
let params = [ idx, idx, 1, id ];
```

### CODE REVIEW

**Comment by XadillaX** [comment]
Anyone who to handle this PR?

**Comment by XadillaX** [comment]
@dougwilson @fengmk2

File: lib/SqlString.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ SqlString.escapeId = function escapeId(val, forbidQualified) {
     return sql;
   }
 
+  if(typeof val === 'number') {
```
Comment (fengmk2) [inline lib/SqlString.js]: style

**Comment by fengmk2** [comment]
Is there any document talk about filed can be a number?

**Comment by XadillaX** [comment]
@fengmk2 cnpm's model layer.

**Comment by dougwilson** [comment]
Sorry, apparently I never actually subscribed to this repository :) I don't see anything wrong with the concept, but I have a few comments on the change itself:

1. What if instead of stringifying only numbers, we just stringified anything that was not an array? Then it would let anything work.
2. The reassigning of an argument variable causes the function to deoptimize in v8. Please assign to a non-argument variable.

**Comment by XadillaX** [comment]
@dougwilson 

For the first point, I don't think some will expect `escapeId(true)` to be \`true`. But if you think it's necessary, I'm glad to stringify anythink was not an array.

For the second one, I will commit later to declare a new variable.

**Comment by dougwilson** [comment]
> For the first point, I don't think someone will expect  escapeId(true)  to be "`true`". But if you think it's necessary, I'm glad to stringify anything was not an array.

Sure, I tend to agree, but i.m.o that same argument would just be instead of even making this PR, just pass the number as a string in the first place... :)

**Comment by XadillaX** [comment]
@dougwilson e.g.

```javascript
for(let i = 0; i < 10; i++) {
  escapeId(i);
}
```

Code above will crashed without this PR. But it's more convenient. Or the code should be like:

```javascript
for(let i = 0; i < 10; i++) {
  escapeId(i.toString());
}
```

**Pull request closed without merging**